src/sympc/protocol/aby3/aby3.py

```diff
@@ -0,0 +1,172 @@
+            security_type : specifies the security level of the Protocol.
+
+        Raises:
+            ValueError : If invalid security_type is provided.
+        """
+        if security_type not in self.security_levels:
+            raise ValueError(f"{security_type} is not a valid security type")
+
+        self.security_type = security_type
+
+    def __eq__(self, other: Any) -> bool:
+        """Check if "self" is equal with another object given a set of attributes to compare.
+
+        Args:
+            other (Any): Object to compare
+
+        Returns:
+            bool: True if equal False if not.
+        """
+        if self.security_type != other.security_type:
+            return False
+
+        if type(self) != type(other):
+            return False
+
+        return True
+
+    @staticmethod
+    def truncation_algorithm1(
+        ptr_list: List[torch.Tensor], shape: torch.Size, session: Session
+    ) -> List[ReplicatedSharedTensor]:
+        """Performs the ABY3 truncation algorithm1.
+
+        Args:
+            ptr_list (List[torch.Tensor]): Tensors to truncate
+            shape(torch.Size) : shape of tensor values
+            session(Session) : session the tensor belong to
+
+        Returns:
+            List["ReplicatedSharedTensor"] : Truncated shares.
+        """
+        rand_value = torch.empty(size=shape, dtype=session.tensor_type).random_(
+            generator=gen
+        )
+        base = session.config.encoder_base
+        precision = session.config.encoder_precision
+        scale = base ** precision
+        x1, x2, x3 = ptr_list
+        x1_trunc = x1 >> precision if base == 2 else x1 // scale
+        x_trunc = (x2 + x3) >> precision if base == 2 else (x2 + x3) // scale
+        shares = [x1_trunc, x_trunc - rand_value, rand_value]
+        ptr_list = ReplicatedSharedTensor.distribute_shares(shares, session)
+        return ptr_list
+
+    @staticmethod
+    def truncate(x: MPCTensor, session: Session) -> MPCTensor:
+        """Perfoms the ABY3 truncation algorithm.
+
+        Args:
+            x (MPCTensor): input tensor
+            session (Session) : session of the input tensor.
+
+        Returns:
+            MPCTensor: truncated MPCTensor.
+
+        Raises:
+            ValueError : parties involved in the computation is not equal to three.
+            ValueError : Invalid MPCTensor share pointers.
+
+        TODO :Switch to trunc2 algorithm  as it is communication efficient.
+        """
+        if session.nr_parties != 3:
+            raise ValueError("Share truncation algorithm 1 works only for 3 parites.")
+
+        # RSPointer - public ops, Tensor Pointer - Private ops
+        ptr_list = []
+        ptr_name = x.share_ptrs[0].__name__
+        if ptr_name == "ReplicatedSharedTensorPointer":
+            ptr_list.append(x.share_ptrs[0].get_shares()[0].get_copy())
+            ptr_list.extend(x.share_ptrs[1].get_copy().shares)
+        elif ptr_name == "TensorPointer":
```

Comment:
Q: Can we reach this case? (won't we always have ```ReplcatedShareTensorPointers```?

Reply:
During `Falcon_multiplication protocol` for private ops we get `Tensor Pointer`,       
https://github.com/OpenMined/SyMPC/blob/99c559bfd82ee9c2f61707fbf258f83f5bad0223/src/sympc/protocol/falcon/falcon.py#L151
We delayed the resharing to reduced round complexity

Reply:
Discussed in slack.